hwilib/devices/tapsigner.py

```diff
@@ -0,0 +1,474 @@
+        """
+        warn = input("This operation will change your card CVC code. "
+                     "Are you sure you want to continue?[y/n]")
+        if warn.lower() == "y":
+            self.interactive_get_cvc()  # 1. ask for current CVC
+            try:
+                # 2. verify current CVC is correct, or fail fast
+                self.device.send_auth("read", self.password, **dict(nonce=pick_nonce()))
+            except (CardRuntimeError, AssertionError):
+                raise ValueError("Invalid CVC")
+            new_cvc = getpass.getpass("New CVC:\n")
+            if not is_cvc_length_range(new_cvc):
+                raise ValueError("CVC has to be min: 6 chars and max: 32 chars long")
+            new_cvc_confirm = getpass.getpass("Confirm new CVC:\n")
+            # 3. confirm that new CVC matches one from confirmation prompt
+            if new_cvc != new_cvc_confirm:
+                raise ValueError("Confirmation failure. CVC mismatch")
+            if self.password == new_cvc:
+                print("New CVC same as old one. NOOP")
+                return False
+            try:
+                # 4. actual CVC change
+                self.device.change_cvc(self.password, new_cvc)  # here it is 100% htat CVC (password is correct) #2
+            except CardRuntimeError as e:
+                if e.code == 425:
+                    raise ValueError("Card not yet backed-up. "
+                                     "Please use 'backup' command first to back-up your card")
+            print("New CVC in effect.")
+            return True
+        return False
+
+    def can_sign_taproot(self) -> bool:
+        """
+        Whether the device has a version that can sign for Taproot inputs
+
+        :return: Whether Taproot is supported
+        """
+        return False
+
+
+def enumerate(password: str = "") -> List[Dict[str, Any]]:
+    results = []
+    devices = find_cards()
+    while True:
+        try:
+            card = next(devices)
+        except StopIteration:
+            break
+        except Exception:
+            continue
+        if not card.is_tapsigner:
+            continue
+        d_data: Dict[str, Any] = {}
+        d_data['path'] = card.card_ident
+        d_data['type'] = 'tapsigner'
+        d_data['model'] = 'tapsigner'
+        d_data['label'] = None
+        d_data['needs_pin_sent'] = False
+        d_data['needs_passphrase_sent'] = False
+
+        if isinstance(card.tr, CKTapUnixTransport):
+            d_data['model'] += '_simulator'
+        if password:
+            # this is kind of unfortunate that we need password (a.k.a CVC) to get master fp
+            # because when using it with core, one has to provide password on the command line
+            # -signer='/home/more/PycharmProjects/HWI/venv/lib/python3.8/site-packages/hwi.py -p 123456'
+            d_data['fingerprint'] = card.get_xfp(password).hex().lower()
+        else:
+            d_data['fingerprint'] = None
```

Comment:
If the password is not provided, this should set an `error` and `code` and omit `fingerprint`, as we do for the digitalbitbox and trezors which have a similar issue.